src/pal/src/exception/signal.cpp

```diff
@@ -430,28 +438,29 @@ Parameters :
 static void inject_activation_handler(int code, siginfo_t *siginfo, void *context)
 {
     // Only accept activations from the current process
-    if (siginfo->si_pid == getpid())
+    if (siginfo->si_pid == getpid() && g_activationFunction != NULL)
```

Comment:
Can you please flip the order of those two things in the test so that we don't waste time calling getpid if we have no activation function?